From babfaa26cf95e8dab8539dc637fee9dc5ffaadc5 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 26 Sep 2018 10:28:38 +0530
Subject: [PATCH 0418/1852] drm: xlnx: sdi: Fix clearing en_st352_c and
 use_ds2_3ga drm properties

commit 11e7368120bad35c8f8555523c2ca632e6622a36 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes the clearing of the drm properties en_st352_c and
use_ds2_3ga. The bits were not cleared when the properties were set to
false.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi.c | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi.c b/drivers/gpu/drm/xlnx/xlnx_sdi.c
index 5e7ca613811a..0e1dd6505ab2 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi.c
@@ -807,11 +807,17 @@ static void xlnx_sdi_setup(struct xlnx_sdi *sdi)
 	if (sdi->enable_anc_data)
 		reg |= XSDI_TX_CTRL_USE_ANC_IN;
 
-	if (sdi->enable_st352_chroma && sdi->en_st352_c_val) {
-		reg |= XSDI_TX_CTRL_INS_ST352_CHROMA;
-
-		if (sdi->use_ds2_3ga_val)
-			reg |= XSDI_TX_CTRL_USE_DS2_3GA;
+	if (sdi->enable_st352_chroma) {
+		if (sdi->en_st352_c_val) {
+			reg |= XSDI_TX_CTRL_INS_ST352_CHROMA;
+			if (sdi->use_ds2_3ga_val)
+				reg |= XSDI_TX_CTRL_USE_DS2_3GA;
+			else
+				reg &= ~XSDI_TX_CTRL_USE_DS2_3GA;
+		} else {
+			reg &= ~XSDI_TX_CTRL_INS_ST352_CHROMA;
+			reg &= ~XSDI_TX_CTRL_USE_DS2_3GA;
+		}
 	}
 
 	xlnx_sdi_writel(sdi->base, XSDI_TX_MDL_CTRL, reg);
-- 
2.31.1

